jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/KubernetesHelper.java

```diff
@@ -952,10 +950,21 @@ public static boolean isControllerResource(HasMetadata h) {
         return Arrays.stream(POD_CONTROLLER_KINDS).anyMatch(c -> c.equals(h.getKind()));
     }
 
-    public static CustomResourceDefinitionContext getCrdContext(CustomResourceDefinitionList customResourceDefinitionList, GenericCustomResource customResource) {
-        return findCrdForCustomResource(customResourceDefinitionList, customResource)
-            .map(CustomResourceDefinitionContext::fromCrd)
+    public static CustomResourceDefinitionContext getCrdContext(APIResourceList customResourceDefinitionList, GenericKubernetesResource customResource) {
+        APIResource apiResource = customResourceDefinitionList.getResources().stream()
+            .filter(r -> r.getSingularName().equals(customResource.getKind().toLowerCase()))
+            .findFirst()
             .orElse(null);
+        if (apiResource != null) {
+            return new CustomResourceDefinitionContext.Builder()
+                .withGroup(ApiVersionUtil.trimGroup(customResource.getApiVersion()))
+                .withVersion(ApiVersionUtil.trimVersion(customResource.getApiVersion()))
+                .withScope(Boolean.TRUE.equals(apiResource.getNamespaced()) ? Scope.NAMESPACED.value() : Scope.CLUSTER.value())
+                .withPlural(apiResource.getName())
+                .withKind(apiResource.getKind())
+                .build();
+        }
+        return null;
```

Comment:
Is this something necessary after https://github.com/fabric8io/kubernetes-client/pull/3525 ??

I think using the KubernetesClient#resource should take care of this.

https://github.com/shawkins/kubernetes-client/blob/e11b8503e8dddc86ff45274ddba1718ec66b9949/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Handlers.java#L64-L125